nav2_behavior_tree/plugins/control/round_robin_node.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "behaviortree_cpp_v3/control_node.h"
+#include "behaviortree_cpp_v3/bt_factory.h"
+
+namespace nav2_behavior_tree
+{
+
+class RoundRobinNode : public BT::ControlNode
+{
+public:
+  explicit RoundRobinNode(const std::string & name)
+  : BT::ControlNode::ControlNode(name, {})
+  {
+    setRegistrationID("RoundRobin");
+  }
+
+  BT::NodeStatus tick() override
+  {
+    const unsigned num_children = children_nodes_.size();
+
+    setStatus(BT::NodeStatus::RUNNING);
+
+    TreeNode * child_node = children_nodes_[current_child_idx_];
+    const BT::NodeStatus child_status = child_node->executeTick();
+
+    switch (child_status) {
+      case BT::NodeStatus::SUCCESS:
+        // Wrap around to the first child
+        if (++current_child_idx_ == num_children) {
+          // TODO(mjeronimo): halt this child (not all children)
```

Comment:
I'd suggest deleting this comment. I don't see a point in just halting this one child. Nothing else can be running at the same time anyhow. Halting everything should be harmless. Halting everything seems to be standard practice in the behavior tree library.

@mjeronimo Is there any particular reason you want to just halt the one child?